### PR TITLE
Change date format #135

### LINKED ISSUE #135: Change date format
Hello!
Can you please change hard coded date format to accept `[params]` format? I'm creating my website with this awesome theme, but in Poland we don't use `January, 2 2006` we use `2 January 2006` and in general it will be better to end user choose their preferred format.
I can make PR with this changes but I'm new to Hugo and don't know how exactly those changes will do to theme in general...
To have same date format like is now you would need to:
In `config.toml` you would need to add:
```
[params]
  dateFormat = "January, 2 2006"
```
in `layouts/partials/list-posts.html` change:
```
<span class="post-date">{{ dateFormat "January 2, 2006" .Date }}</span>
```
to:
```
<span class="post-date"> {{ dateFormat .Site.Params.dateFormat .Date }} </span>
```
and in: `layouts/partials/post.html` change:
```
{{ dateFormat "January 2, 2006" .Date }}
```
to:
```
 {{ dateFormat .Site.Params.dateFormat .Date }}
```

### PR DESCRIPTION
**Pull request by mansoorbarri**
fixes #135

### CODE REVIEW

**Comment by yashmehrotra** [comment]
@mansoorbarri Thanks for the PR

Can you also:
1. Document this in theme basics page
2. Use the current one as default fallback so that we don't have to put that in config.toml

**Comment by mansoorbarri** [comment]
@yashmehrotra updated

File: layouts/partials/list-posts.html [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <article class="post-title">
     <a href="{{ .Permalink }}" class="post-link">{{ .Title }}</a>
     <div class="flex-break"></div>
-    <span class="post-date">{{ dateFormat "January 2, 2006" .Date }}</span>
+    <span class="post-date">
+        {{ if .Site.Params.dateFormat }} 
+            {{ dateFormat .Site.Params.dateFormat .Date }}
+        {{ else }}
+            {{ dateFormat "January 2, 2006" .Date }} 
+        {{ end }} </span>
```
Comment (yashmehrotra) [inline layouts/partials/list-posts.html:9]:
```suggestion
    <span class="post-date">{{ dateFormat (or .Site.Params.dateFormat "January 2, 2006") }}</span>
```

I think this looks much cleaner, can you try with this

File: layouts/partials/post.html [hunk 2]
```diff
@@ -3,7 +3,12 @@
         <h1>{{ .Title }}</h1>
         <small role="doc-subtitle">{{ .Description }}</small>
         <p class="post-date">
-            {{ dateFormat "January 2, 2006" .Date }}{{ if lt .Date .Lastmod }} | Updated {{ dateFormat "January 2, 2006" .Lastmod }}{{ end }}
+            {{ if .Site.Params.dateFormat }} 
+            {{ dateFormat .Site.Params.dateFormat .Date }}
+        {{ else }}
+            {{ dateFormat "January 2, 2006" .Date }} 
+        {{ end }}
+        {{ if lt .Date .Lastmod }} | Updated {{ dateFormat .Site.Params.dateFormat .Lastmod }}{{ end }}
```
Comment (yashmehrotra) [inline layouts/partials/post.html:11]: Same as above

File: exampleSite/content/posts/theme-documentation-basics.md [hunk 3]
```diff
@@ -213,6 +213,14 @@ If you want to display posts on the homepage, the options are:
   showPostsOnHomePage = "popular"
 ```
 
+### Date format 
+Change date format
```
Comment (yashmehrotra) [inline exampleSite/content/posts/theme-documentation-basics.md:217]: Change to:
The date format being used across can be chaged. For valid date strings see: https://pkg.go.dev/time#pkg-constants

File: layouts/partials/list-posts.html (hunk 1 above)
Comment (mansoorbarri) [reply layouts/partials/list-posts.html]: > I think this looks much cleaner, can you try with this

too true, I have updated the code

**Comment by yashmehrotra** [comment]
Thanks, merged

**Pull request merged**